src/panels/config/backup/components/ha-backup-details-restore.ts

```diff
@@ -0,0 +1,145 @@
+import { css, html, LitElement } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import "../../../../components/ha-card";
+import "../../../../components/ha-md-list";
+import "../../../../components/ha-md-list-item";
+import "../../../../components/ha-button";
+import "./ha-backup-data-picker";
+import type { HomeAssistant } from "../../../../types";
+import type { LocalizeFunc } from "../../../../common/translations/localize";
+import type {
+  BackupContentExtended,
+  BackupData,
+} from "../../../../data/backup";
+import { fireEvent } from "../../../../common/dom/fire_event";
+
+declare global {
+  interface HASSDomEvents {
+    "backup-restore": { selectedData?: BackupData };
+  }
+}
+
+@customElement("ha-backup-details-restore")
+class HaBackupDetailsRestore extends LitElement {
+  @property({ attribute: false }) public hass?: HomeAssistant;
+
+  @property({ attribute: false }) public localize!: LocalizeFunc;
+
+  @property({ type: Object }) public backup!: BackupContentExtended;
+
+  @property({ type: Boolean, attribute: "ha-required" })
+  public haRequired = false;
+
+  @property({ attribute: "translation-key-panel" }) public translationKeyPanel:
+    | "page-onboarding.restore"
+    | "config.backup" = "config.backup";
+
+  @state() private _selectedData?: BackupData;
+
+  protected willUpdate() {
+    if (!this.hasUpdated && this.haRequired) {
+      this._selectedData = {
+        homeassistant_included: true,
+        folders: [],
+        addons: [],
+        homeassistant_version: this.backup.homeassistant_version,
+        database_included: this.backup.database_included,
+      };
+    }
+  }
+
+  render() {
+    return html`
+      <ha-card>
+        <div class="card-header">
+          ${this.localize(
+            `ui.panel.${this.translationKeyPanel}.details.restore.title`
+          )}
+        </div>
+        <div class="card-content">
+          <ha-backup-data-picker
+            .translationKeyPanel=${this.translationKeyPanel}
+            .localize=${this.localize}
+            .hass=${this.hass}
+            .data=${this.backup}
+            .value=${this._selectedData}
+            @value-changed=${this._selectedBackupChanged}
+            .requiredItems=${this.haRequired ? ["config"] : []}
```

Comment:
Dont create arrays in render, as they will always change it will cause a re-render down the line on every render. Wrap it in a memoize